iroh/src/magicsock.rs

```diff
@@ -290,12 +303,15 @@ impl MagicSock {
         // send this.  Before the handshake has completed we don't have anything useful to
         // do with this connection inside of the EndpointStateActor anyway.
         let weak_handle = conn.weak_handle();
-        let endpoint_state = self.endpoint_map.endpoint_state_actor(remote);
-        let msg = EndpointStateMessage::AddConnection(weak_handle, paths_info);
+        let (sender, selected_path) = self
```

Comment:
I'm going to try complaining about this bit once again :)

You're introducing a 2nd way of passing a watcher/watchable to/from the actor.  The selected path is passed from the actor to here while the open_paths is passed the other way round.

I'm not particularly fond of how open_paths is hooked up, I considered it a hack when I did it as well.  But it was an attempt to keep things in the sync world that could stay there.  I prefer it slightly over the way you just introduced because we don't need to push this selected_path watcher through the handle and have different constructor methods for the actor?  In either case, having two ways of doing this is a bit weird?

Reply:
Difference is that the selected_path is per EndpointStateActor while open_paths is per Connection. I also didn't really like my solution, I'll see if I can come up with something better.